pkg/compiler/builders.go

```diff
@@ -22,9 +22,10 @@ type workflowBuilder struct {
 	// These are references to all subgraphs and tasks passed to CompileWorkflow. They will be passed around but will
 	// not show in their entirety in the final Graph. The required subset of these will be added to each subgraph as
 	// the compile traverses them.
-	allLaunchPlans  map[string]c.InterfaceProvider
-	allTasks        c.TaskIndex
-	allSubWorkflows c.WorkflowIndex
+	allLaunchPlans          map[string]c.InterfaceProvider
+	allTasks                c.TaskIndex
+	allSubWorkflows         c.WorkflowIndex
+	allCompiledSubWorkflows c.WorkflowIndex
```

Comment:
interesting, this looks like a big change ha?

Reply:
Not quite, it used to keep the non-compiled workflows in allSubworkflows and when they get compiled, we replace them in-place. Now they live in two separate lists so we know which ones we compiled before...

Reply:
aah ok, this is better :)